src/lib/logical_query_plan/lqp_utils.hpp

```diff
@@ -165,4 +165,18 @@ void visit_lqp_upwards(const std::shared_ptr<AbstractLQPNode>& lqp, Visitor visi
  */
 std::vector<std::shared_ptr<AbstractLQPNode>> lqp_find_subplan_roots(const std::shared_ptr<AbstractLQPNode>& lqp);
 
+/**
+ * @return A set of column expressions created by the given @param lqp_node, matching the given @param column_ids.
+ *         Currently, only StoredTableNode, StaticTableNode and MockNode are supported.
```

Comment:
This is a helper method that maps column ids from tables to the matching output expressions. Conceptually, it can only work on data source nodes. Currently, these are StoredTableNodes, StaticTableNodes and MockNodes.